app/assets/javascripts/controllers/catalog/catalog_item_form_controller.js

```diff
@@ -95,26 +95,22 @@ ManageIQ.angular.app.controller('catalogItemFormController', ['$scope', '$timeou
       return;
     }
 
-    if (data.key) {
-      $timeout(function() {
-        // get ids of checked additional tenants in the tree
-        var new_ids = miqTreeObject('tenants_tree').getChecked().map(getTenantId);
-
-        // filter unnecessary root node if present
-        new_ids = new_ids.filter(function(tenant_id) {
-          return tenant_id;
-        });
-
-        if (new_ids) {
-          vm.catalogItemModel.additional_tenant_ids = new_ids.sort();
-        }
-      });
-    }
+    $timeout(function() {
+      // get ids of all checked tenants in the tree
+      var new_ids = Object.values(ManageIQ.redux.store.getState().tenants_tree).filter(
```

Comment:
this looks like

```
var new_ids = miqGetSelectedKeys(ManageIQ.redux.store.getState().tenants_tree).map(getTenantId);
```

(+ fix `getTenantId` to expect the key directly)

(Also, is this really the only place using a "give me all the selected items in the current tree" logic?)